bbb-common-web/src/main/java/org/bigbluebutton/presentation/imp/PresentationProcessExternal.java

```diff
@@ -0,0 +1,185 @@
+    private int maxImageWidth = 2048;
+    private int maxImageHeight = 1536;
+    private int placementsThreshold = 800;
+    private int imageTagThreshold = 800;
+    private int pngScaleTo = 800;
+    private int pngDefaultWidth = 500;
+    private int pngDefaultHeight = 500;
+    private int pdfFontsTimeout = 3;
+    private int maxNumberOfAttemptsForPdfFonts = 3;
+    private int svgPresentationResolutionPpi = 300;
+    private boolean forceRasterizeSlides = false;
+    private String pngWidthRasterizedSlides = "2048";
+    private String pageConversionMemoryHigh = "512M";
+    private String pageConversionMemoryMax = "640M";
+
+
+    public boolean processPage(String meetingId, String presentationId, int page) {
+        List<String> command = Arrays.asList(
+                "systemd-run", "--user", "--pipe", "--wait", "--same-dir",
+                "-p", "Environment=PRESENTATIONS_DIR=" + presentationDir,
+                "-p", "Environment=THUMBNAIL_SCALE=" + thumbScale,
+                "-p", "Environment=PDF_FONTS_TIMEOUT=" + pdfFontsTimeout,
+                "-p", "Environment=PDF_FONTS_MAX_ATTEMPTS=" + maxNumberOfAttemptsForPdfFonts,
+                "-p", "Environment=SVG_RESOLUTION_PPI=" + svgPresentationResolutionPpi,
+                "-p", "Environment=MAX_SVG_PATHS=" + placementsThreshold,
+                "-p", "Environment=MAX_SVG_IMGS=" + imageTagThreshold,
+                "-p", "Environment=MAX_IMAGE_WIDTH=" + maxImageWidth,
+                "-p", "Environment=MAX_IMAGE_HEIGHT=" + maxImageHeight,
+                "-p", "Environment=PNG_DEFAULT_WIDTH=" + pngDefaultWidth,
+                "-p", "Environment=PNG_DEFAULT_HEIGHT=" + pngDefaultHeight,
+                "-p", "Environment=PNG_SCALE_TO=" + pngScaleTo,
+                "-p", "Environment=GENERATE_PNGS=" + booleanToString(generatePngs),
+                "-p", "Environment=RASTERIZE_SLIDE_FORCE=" + booleanToString(forceRasterizeSlides),
+                "-p", "Environment=RASTERIZE_PNG_WIDTH=" + pngWidthRasterizedSlides,
+                "-p", "ProtectSystem=strict",
+                "-p", "ProtectHome=yes",
+                "-p", "PrivateTmp=true",
+                "-p", "PrivateUsers=true",
+                "-p", "NoNewPrivileges=true",
+                "-p", "RestrictRealtime=true",
+                "-p", "SystemCallFilter=~@mount",
+                "-p", "Type=oneshot",
+                "-p", "TimeoutSec=" + (maxConversionTime * 60),
+                "-p", "MemoryHigh=" + pageConversionMemoryHigh,
+                "-p", "MemoryMax=" + pageConversionMemoryMax,
+                "-p", "MemorySwapMax=0",
+                "/usr/local/bin/bbb-process-pdf-page.sh",
+                meetingId + "_" + presentationId +"_" + page
+        );
+
+
+        log.debug("Starting processing page [{}]", String.join(" ", command));
+        return new ExternalProcessExecutor().exec(command, Duration.ofMillis(maxConversionTime * 60 * 1000));
```

Comment:
What does ExternalProcessExecutor do when this duration is exceeded? How does that interact with systemd-run?

The duration configured here should probably be somewhat longer than the TimeoutSec on the systemd-run command.

Reply:
Looks like it stops waiting but actually the process keeps running.
I increased the timeout to be 1 second higher than the `systemd-run TimeoutSec=`